adobe_xd/lib/pinned.dart

```diff
@@ -216,43 +198,40 @@ class RenderPinned extends RenderShiftedBox {
   Pin _hPin;
   Pin _vPin;
 
-  RenderPinned({Pin hPin, Pin vPin, RenderBox child})
-      : assert(hPin != null),
-        assert(vPin != null),
-        _hPin = hPin,
+  RenderPinned({required Pin hPin, required Pin vPin, RenderBox? child})
+      : _hPin = hPin,
         _vPin = vPin,
         super(child);
 
   _Span _calculateSpanFromPin(Pin pin, double maxSize) {
```

Comment:
There are too many ! operators here, it's basically side-stepping nnbd completely.

This might work better if _calculateSpanFromPin declares the non-null dependancies it needs, 
```dart
_calculateSpanFromPin(double maxSize, double pinStart, double pinEnd etc)
```

Then, when  called, you can provide inline defaults: 
```dart
_calculateSpanFromPin(maxSize, pin.start ?? 0, pin.end ?? 0, etc ... )
```

If that creates too much boilerplate on the call side, you could use inline defaults inside the method: `pin.start ?? 0` or create local vars to hold them: `double start = pin.start ?? 0`

Reply:
I think it's due to all the assertions here:    https://github.com/jrc2139/xd-to-flutter-plugin/blob/b6e5f39512a5994b8e3ebc317ce601eb503631be/adobe_xd/lib/pinned.dart#L166